webapp/ng_tests/controllers/main_page_controller_test.js

```diff
@@ -2,11 +2,12 @@
 
 describe('MainPageController', function () {
     beforeEach(module('lunchr'));
+    beforeEach(module('stateMock')); //mock the $state service of ui-router
 
     var DEFAULT_EMAIL = 'user';
     var DEFAULT_PASSWORD = 'password';
 
-    var $controller, $httpBackend, $rootScope, $location;
+    var $controller, $httpBackend, $rootScope, $state;
```

Comment:
Before, we would navigate from page to page using $location which would alter the url which angular would pick up.

Now, $state tells angular to go to a specific state which then gets reflected by a URL change.